thavalon-webapp/src/utils/GameSocket.tsx

```diff
@@ -0,0 +1,67 @@
+enum WEBSOCKET_READYSTATE {
+    CONNECTING = 0,
+    OPEN = 1,
+    CLOSING = 2,
+    CLOSED = 3,
+};
+
+
+interface PingResponse {
+    "result": boolean,
+    "errorMessage": string, // only populated if result is false
+}
+
+export class GameSocket {
+    // the instantiated instance of the gamesocket
+    private static instance: GameSocket;
+    // the instantiated instance of the underlying websocket
+    private static websocket: WebSocket;
+
+    // functions for handling incoming websocket events
+    private static socketOnOpen(event: Event) {
+        console.log("Successfully initiated connection.");
+    }
+
+    private static socketOnMessage(event: MessageEvent) {
+        console.log(event);
+        console.log("Received message: " + event.data);
+    }
+
+    private static socketOnClose(event: CloseEvent) {
+        console.log("Recieved on close message.");
+    }
+
+    private static socketOnError(event: Event) {
+        console.log("Received on error message.");
+    }
+
+    /**
+     * Get the account manager instance, creating one if needed.
+     * 
+     * @param socketUrl The url to connect to for the websocket
+     * @returns The instance of the AccountManager.
+     */
+    public static getInstance(socketUrl: string): GameSocket {

```

Comment:
tioli, but you don't need public in Typescript. Things are public by default.

Reply:
Also, should we force getInstance to take a socket URL?  If we ever need to get an instance elsewhere, we'll need to have the socketURL. Maybe a different function to take the URL?

Reply:
personally I like having public, makes it easier to read imo. What about a getOrCreate instance that optionally takes socketUrl? If socketUrl provided, will make a new websocket (and close an existing one).